src/iexamine.cpp

```diff
@@ -5936,36 +5936,80 @@ void iexamine::mill_finalize( Character &, const tripoint &examp, const time_poi
         return;
     }
 
-    for( map_stack::iterator iter = items.begin(); iter != items.end(); ) {
+
+    // Get list of milable items, their rot timer and their flags
+    std::map<itype_id, std::vector<double>> millable_rot;
+    std::map<itype_id, std::vector<flag_id>> millable_flags;
+    std::map<itype_id, item> millable_items;
+    std::vector<map_stack::iterator> iter_to_delet;
+    for( map_stack::iterator iter = items.begin(); iter != items.end(); ++iter ) {
         item &it = *iter;
         if( it.type->milling_data ) {
             it.calc_rot_while_processing( milling_time );
-            const islot_milling &mdata = *it.type->milling_data;
-            const int resulting_charges = get_milled_amount( it.typeId(), examp, here );
-            // if not enough material, just remove the item (0 loops)
-            // (may happen if the player did not add enough charges to the mill
-            // or if the conversion rate is changed between versions)
-            for( int i = 0; i < resulting_charges; i++ ) {
-                item result( mdata.into_, start_time + milling_time );
-                result.components.add( it );
-                // copied from item::inherit_flags, which can not be called here because it requires a recipe.
+
+            if( millable_rot.find( it.typeId() ) == millable_rot.end() ) {
+                const std::vector<double> rot_times = { it.get_relative_rot() };
+                millable_rot.emplace( it.typeId(), it.get_relative_rot() );
```

Comment:
`it.get_relative_rot()` here always returns 0 nad I have no idea why

Reply:
`it.calc_rot_while_processing( milling_time );` on line 5948 is setting the rot ahead of the current time (to the end of the processing duration)

Reply:
When I tried commenting it away get_relative_rot was still giving me 0, and it should make things more rotten not less in any case, no?

Reply:
It might be that putting items into the mill already loses the rot data? Though I couldn't find anything obvious at a glance.

Reply:
Probably because the mill is processing the millable, meaning the `goes_bad()` check in `get_relative_rot()` (item.cpp@7806?) returns false.

Probably this is why there is this part in the original:
```c
// Set flag to tell set_relative_rot() to calc from bday not now
result.set_flag( flag_PROCESSING_RESULT );
result.set_relative_rot( it.get_relative_rot() );
result.unset_flag( flag_PROCESSING_RESULT );
```

Reply:
removing the flag before getting relative rot doesn't solve the issue

Reply:
Okay, that was my best guess (which doesn't mean much tbh) but I'll take another look.

Reply:
Well, as far as I can see `relative_rot` is `rot / get_shelf_life()` and `rot` is only set in `set_rot` and `set_relative_rot` otherwise `last_temp_check` is used. Does that help?

Reply:
My bad! Removing `flag_PROCESSING` seems to solve the issue, I was messing with flag_PROCESSING_RESULT which is not the same thing.